alpha/declcfg/write.go

```diff
@@ -3,13 +3,100 @@ package declcfg
+// 	      neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.3["neuvector-operator.v1.3.3"]-- replaces --> neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.2["neuvector-operator.v1.3.2"]
+// 	      neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.3["neuvector-operator.v1.3.3"]-- skips --> neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.2.9["neuvector-operator.v1.2.9"]
+// 	      neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.4["neuvector-operator.v1.3.4"]
+// 	      neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.4["neuvector-operator.v1.3.4"]-- replaces --> neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.3["neuvector-operator.v1.3.3"]
+// 	      neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.3.4["neuvector-operator.v1.3.4"]-- skips --> neuvector-certified-operator-rhmp-beta-neuvector-operator.v1.2.9["neuvector-operator.v1.2.9"]
+//     end
+//   end
+// end
+func WriteMermaidChannels(cfg DeclarativeConfig, out io.Writer) error {
+	pkgs := map[string]*strings.Builder{}
+
+	sort.Slice(cfg.Channels, func(i, j int) bool {
+		return cfg.Channels[i].Name < cfg.Channels[j].Name
+	})
+
+	for _, c := range cfg.Channels {
+		pkgBuilder, ok := pkgs[c.Package]
+		if !ok {
+			pkgBuilder = &strings.Builder{}
+			pkgs[c.Package] = pkgBuilder
+		}
+		channelID := fmt.Sprintf("%s-%s", c.Package, c.Name)
+		pkgBuilder.WriteString(fmt.Sprintf("    %%%% channel %q\n", c.Name))
+		pkgBuilder.WriteString(fmt.Sprintf("    subgraph %s[%q]\n", channelID, c.Name))
+
+		for _, ce := range c.Entries {
+			entryId := fmt.Sprintf("%s-%s", channelID, ce.Name)
+			pkgBuilder.WriteString(fmt.Sprintf("      %s[%q]\n", entryId, ce.Name))
+
+			// no support for SkipRange yet
+			if len(ce.Replaces) > 0 {
+				replacesId := fmt.Sprintf("%s-%s", channelID, ce.Replaces)
+				pkgBuilder.WriteString(fmt.Sprintf("      %s[%q]-- %s --> %s[%q]\n", entryId, ce.Name, "replaces", replacesId, ce.Replaces))
+			}
+			if len(ce.Skips) > 0 {
+				for _, s := range ce.Skips {
+					skipsId := fmt.Sprintf("%s-%s", channelID, s)
+					pkgBuilder.WriteString(fmt.Sprintf("      %s[%q]-- %s --> %s[%q]\n", entryId, ce.Name, "skips", skipsId, s))
+				}
+			}
+		}
+		pkgBuilder.WriteString("    end\n")
+	}
+
+	out.Write([]byte("<!-- PLEASE NOTE:  skipRange edges are not currently displayed -->\n"))
+	out.Write([]byte("graph LR\n"))
+	pkgNames := []string{}
+	for pname, _ := range pkgs {
+		pkgNames = append(pkgNames, pname)
+	}
+	sort.Slice(pkgNames, func(i, j int) bool {
+		return pkgNames[i] < pkgNames[j]
+	})
+	for _, pkgName := range pkgNames {
+		out.Write([]byte(fmt.Sprintf("  %%%% package %q\n", pkgName)))
+		out.Write([]byte(fmt.Sprintf("  subgraph %q\n", pkgName)))
+		out.Write([]byte(pkgs[pkgName].String()))
+		out.Write([]byte("  end\n"))
+	}
+	out.Write([]byte("<!-- PLEASE NOTE:  skipRange edges are not currently displayed -->\n"))
```

Comment:
```suggestion
	out.Write([]byte("%% PLEASE NOTE:  skipRange edges are not currently displayed\n"))
```

Reply:
As we discussed out-of-band, the comment is designed to break mermaid renderers so the user is forced to ack the limitation. 
Added a func comment to that effect so are are good in future.